OpenRA.Game/Game.cs

```diff
@@ -189,8 +193,47 @@ public static void RestartGame()
 			// Restart the game with the same replay/mission
 			if (replay != null)
 				JoinReplay(replayName);
-			else
+			else if (mission)
 				StartMission(uid, globalSettings.GameSpeedType, globalSettings.Difficulty);
+			else
+			{
+				Settings.Server.Map = map;
+				Settings.Save();
+
+				JoinServer(IPAddress.Loopback.ToString(), CreateLocalServer(map), "");
+				OpenWindow("SERVER_LOBBY", new WidgetArgs
+				{
+					{ "onExit", () =>
+						{
+							Disconnect();
+							LoadShellMap();
+						}
+					},
+					{ "onStart",
+						() => { }
+					},
+					{ "skirmishMode", true }
+				});
+
+				// Needs to run after delay to overwrite skirmish default bot addition
+				RunAfterDelay(200, () =>
```

Comment:
I could change this to 
`LobbyInfoChanged += WidgetUtils.Once(() =>` if I set skirmishMode to false however that changes the back button to leave instead of back